src/test_l3_dsHdmiIn.c

```diff
@@ -0,0 +1,1396 @@
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_get_inputports(void)
+{
+    gTestID = 2;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    uint8_t numInputs = 0;
+    dsError_t ret   = dsERR_NONE;
+
+    UT_LOG_INFO("Calling dsHdmiInGetNumberOfInputs(OUT:numInputs:[ ])");
+
+    ret = dsHdmiInGetNumberOfInputs(&numInputs);
+
+    UT_LOG_INFO("Result dsHdmiInGetNumberOfInputs(OUT:numInputs:[%d])",numInputs);
+    DS_ASSERT(ret == dsERR_NONE);
+
+    if(!numInputs)
+       UT_LOG_INFO("Result: Platform does not supports HdmiIn Ports");
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test get number of input ports.
+*
+* This test function gets HdmiInput ports on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 003@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_get_status(void)
+{
+    gTestID = 3;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t ret   = dsERR_NONE;
+    dsHdmiInStatus_t inputstatus = {0};
+    
+    UT_LOG_INFO("Calling dsHdmiInGetStatus(OUT:dsHdmiInStatus_t:[])");
+
+    ret = dsHdmiInGetStatus(&inputstatus);
+
+    UT_LOG_INFO("Result dsHdmiInGetStatus(OUT:inputstatus:[isPresented:[%s], activeport:[%s] ])",
+                 UT_Control_GetMapString(bool_mapTable, inputstatus.isPresented),
+                 UT_Control_GetMapString(dsHdmiInPort_mapTable, inputstatus.activePort));
+    DS_ASSERT(ret == dsERR_NONE);
+
+    for(int i = 0 ; i < dsHDMI_IN_PORT_MAX ; i++) {
+
+           UT_LOG_INFO("Result dsHdmiInGetStatus(OUT:inputstatus:[ port:[%s]:[%d], isPortConnected:[%s])",
+                         UT_Control_GetMapString(dsHdmiInPort_mapTable, i),i,
+                         UT_Control_GetMapString(bool_mapTable, inputstatus.isPortConnected[i]));
+    }
+
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief This test to select port.
+*
+* This test function selects HdmiInput port on platform.
+*
+* **Test Group ID:** 03@n
+* **Test Case ID:** 004@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation
+* [ds-hdmi-in_halSpec.md](../../docs/pages/ds-hdmi-in_halSpec.md)
+*/
+void test_l3_HdmiIn_select_port(void)
+{
+    gTestID = 4;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+    
+    dsError_t ret   = dsERR_NONE;
+    dsHdmiInPort_t port = dsHDMI_IN_PORT_MAX;
+    int32_t select = 0; 
+    bool audmix = false;
+    dsVideoPlaneType_t videoplanetype = dsVideoPlane_MAX;
+    bool topmost = false;
+    uint8_t numInputPorts = 0;
+ 
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/numberOfPorts");
+
+    UT_LOG_INFO(" \n================Please Select Inputs==================\n");
+    UT_LOG_INFO("\n*******Enter the port number to select*******\n");
```

Comment:
Here, check whether we can use **"UT_LOG_MENU_INFO("** instead of **UT_LOG_INFO(**

Please check and update this case wherever needed. Mostly on getting inputs.

Reply:
Is # define required ?

Reply:
done